### PR TITLE
Add support for a user-friendly Windows installer

### PR DESCRIPTION
**Pull request by InvisibleRain**
The installer being used here is [NSIS](https://nsis.sourceforge.io/Screenshots)
image

Also, the daemon was wrapped in [WinSW](https://github.com/winsw/winsw) to ease integration with Windows services

### CODE REVIEW

**Comment by mzz2017** [comment]
Great! Bundled v2ray-core is acceptable.

I'll take a test before merging it.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -185,6 +185,28 @@ jobs:
...
+        ARCH: x64
+      run: |
+        mkdir -p /tmp/NSIS/v2rayA && cd /tmp/NSIS
+        cp $GITHUB_WORKSPACE/LICENSE $PWD/License.txt
+        cp $GITHUB_WORKSPACE/gui/public/img/icons/windows-icon-255x255.ico v2rayA/v2raya.ico
+        cp $GITHUB_WORKSPACE/gui/public/img/icons/windows-bitmap-255x255.bmp v2rayA/v2raya.bmp
+        cp $GITHUB_WORKSPACE/install/windows/v2raya.xml v2rayA/
+        cp $GITHUB_WORKSPACE/install/windows/v2raya.nsi $PWD
+        cp $P_DIR/v2raya_windows_x64_${{ steps.prep.outputs.VERSION }}.exe v2rayA/
+        wget https://github.com/v2fly/v2ray-core/releases/download/v4.44.0/v2ray-windows-64.zip
```
Comment (Gaojianli) [inline .github/workflows/release.yml:201]: Should we replace `v4.44.0` with `latest`? The `v2ray-core` is keep updating.

Comment (mzz2017) [reply .github/workflows/release.yml:201]: It should be modified after the release of v5, but not now, I think.

Comment (Gaojianli) [reply .github/workflows/release.yml:201]: > It should be modified after the release of v5, but not now, I think.

Means that we don't follow the `v4.45.0`(if it exists)?

**Comment by mzz2017** [comment]
@InvisibleRain 
Most things seem good.
But I found a problem. Could you please set the argument `--log-file` for v2raya? It allows we check the log in web. See https://github.com/v2rayA/v2rayA/pull/379.

image

To make the function work, the `--log-file` should be given.

File: .github/workflows/release.yml (hunk 1 above)
Comment (mzz2017) [reply .github/workflows/release.yml:201]: @Gaojianli We could follow it by manual in this duration.

**Comment by InvisibleRain** [comment]
That should do the trick

**Comment by mzz2017** [comment]
Thanks. It works now.
But when the v2rayA is already installed, the updating (install with the newer version installer) will failed with "Could not open file for writing".
Could you please resolve this problem?

**Comment by mzz2017** [comment]
Is there any way to uninstall it before installing?

**Comment by InvisibleRain** [comment]
Done. Is there anything else I might need to fix or improve?

**Comment by mzz2017** [comment]
This version is quiet good now.

Nevertheless, if you want to improve it, the improvement can be made includes:

image

1. Place a shortcut of Uninstall here.
2. Write installation info in regedit bbecause user may choose a different installation location which may cause failure to uninstall the old one. See [Auto-uninstall old before installing new](https://nsis.sourceforge.io/Auto-uninstall_old_before_installing_new).

However, it is okay to merge if you want, and I'll relay you to perfect it.

**Comment by InvisibleRain** [comment]
>Place a shortcut of Uninstall here.

Done


>Write installation info in regedit bbecause user may choose a different installation location which may cause failure to uninstall 
the old one. See Auto-uninstall old before installing new.

Generally, it's a good idea, but I failed to accomplish it. The code from "Auto-uninstall old before installing new" wouldn't launch `.onInit` for some reason and debugging a function with variables named like `$0` `$1` `$2` isn't trivial to say the least


>Arm64 support.

This sound [possible](http://forums.winamp.com/showthread.php?t=455154). But since there are no official arm64 builds for NSIS and WinSW, this would complicate the CI process tremendously for very little profit. Furthermore, I don't have an arm64 Windows machine, so this would complicate manual testing even further. Personally, I would wait to a point when Windows on arm becomes more popular and such core projects like NSIS and WinSW start publishing official arm releases


You can go ahead and merge the pull request at this point

**Comment by mzz2017** [comment]
Thank you for your tireless efforts and awesome work!

**Pull request merged**